packages/flutter/lib/src/rendering/layer.dart

```diff
@@ -16,7 +16,7 @@ import 'debug.dart';
 /// During painting, the render tree generates a tree of composited layers that
 /// are uploaded into the engine and displayed by the compositor. This class is
 /// the base class for all composited layers.
-abstract class Layer {
+abstract class Layer extends TreeDiagnostics {
```

Comment:
RenderObject has a much more complicated tree but it uses the same API, so maybe we can mix in TreeDiagnostics and then just override the part that needs overriding to get the effect we have now.